livingpark_utils/download/ppmi.py

```diff
@@ -0,0 +1,270 @@
+            Tuple with the successful and missing raw data identifiers, respectlively.
+        """
+        cohort = query
+        missing_patno = cohort["PATNO"]
+        try:
+            ppmi_dl = ppmi_downloader.PPMIDownloader()
+            print(f"Downloading image data of {missing_patno.nunique()} subjects")
+            ppmi_dl.download_imaging_data(
+                missing_patno.unique(),
+                type="nifti",
+                timeout=timeout * missing_patno.nunique(),
+                headless=self.headless,
+            )
+        except Exception:
+            missing = self.missing_raw_data(query)
+            success = query[~query["PATNO"].isin(missing["PATNO"])]
+            return success, missing
+
+        # Find cohort file names among downloaded files
+        results_path = "outputs"
+        ppmi_fd = ppmi_downloader.PPMINiftiFileFinder()
+        for _, row in cohort.iterrows():
+            if not row["File name"] or row["File name"] is None:
+                filename = ppmi_fd.find_nifti(
+                    row["PATNO"], row["EVENT_ID"], row["Description"]
+                )
+                if filename is None:
+                    print(
+                        "Not found: "
+                        + f"{row['PATNO'], row['EVENT_ID'], row['Description']}"
+                    )
+                else:  # copy file to dataset
+                    dest_dir = os.path.join(
+                        "inputs",
+                        f'sub-{row["PATNO"]}',
+                        f'ses-{row["EVENT_ID"]}',
+                        "anat",
+                    )
+                    os.makedirs(dest_dir, exist_ok=True)
+                    dest_file = os.path.join(dest_dir, os.path.basename(filename))
+                    os.rename(filename, dest_file)
+                    row["File name"] = dest_file
+
+        # Update file names in cohort
+        cohort["File name"] = cohort.apply(
+            lambda x: ppmi.find_nifti_file_in_cache(
+                x["PATNO"], x["EVENT_ID"], x["Description"]
+            ),
+            axis=1,
+        )
+
+        # Create symlinks to inputs if necessary
+        if symlink:
+            for file_name in cohort["File name"].values:
+                dest_dir = os.path.dirname(file_name).replace(
+                    os.path.join(self.cache_dir, "inputs"),
+                    os.path.join(results_path, "pre_processing"),
+                )
+                dest_file = os.path.join(
+                    dest_dir,
+                    os.path.basename(file_name.replace(self.cache_dir, "")),
+                )
+                if not os.path.exists(dest_file):
+                    os.makedirs(dest_dir, exist_ok=True)
+                    os.symlink(
+                        os.path.relpath(os.path.abspath(file_name), start=dest_file),
+                        dest_file,
+                    )
+
+        return query, pd.DataFrame()
+
+    def missing_raw_data(
```

Comment:
raw data is ambiguous, it should be `T1_nifti_file`

Reply:
Fixed in new commit